api/v2/compat.go

```diff
@@ -180,8 +181,15 @@ func OpenAPIAlertsToAlerts(apiAlerts open_api_models.PostableAlerts) []*types.Al
 // ModelLabelSetToAPILabelSet converts prometheus_model.LabelSet to open_api_models.LabelSet.
 func ModelLabelSetToAPILabelSet(modelLabelSet prometheus_model.LabelSet) open_api_models.LabelSet {
 	apiLabelSet := open_api_models.LabelSet{}
+
+	keys := make([]string, 0, len(modelLabelSet))
 	for key, value := range modelLabelSet {
-		apiLabelSet[string(key)] = string(value)
+		keys = append(keys, string(key))
+		_ = value
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		apiLabelSet[string(key)] = string(modelLabelSet[prometheus_model.LabelName(key)])
```

Comment:
Hi! 👋 Thanks for opening a PR, however I'm not sure this works. `apiLabelSet` is a map which is not sorted, so while you have sorted the keys, when you put them back into `apiLabelSet` they become unsorted again.

Reply:
Yes but when `apiLabelSet` is JSON encoded it will be done so in random order because in Go maps are not sorted.

Reply:
tks for your prompt reply, that's great, but I don't quite understand your question.
If the `apiLabelSet`  map is  sorted, the JSON encoded results are also sorted, in the same order as the map。
As for the randomed `modelLabelSet`, we don't need to pay attention.

Reply:
> If the apiLabelSet map is sorted, the JSON encoded results are also sorted, in the same order as the map

That's the issue 🙂 The `apiLabelSet` map is not sorted. Even though you are inserting keys in a sorted order, they get unsorted by the internal hash function within the map. You cannot sort maps in Go.

Reply:
My suggestion is if you want them to appear sorted in the UI the best place to do it is in the Elm code. You can find it in `ui/app`.

Reply:
Go's map is not sorted, but if the key of the map is of string type, it will be sorted by key when using json.Marshal encoding, so there is no need to sort the map

Reply:
`json.Marshal` sorts maps – I learned something! 😄